apps/api/views.py

```diff
@@ -305,3 +324,61 @@ def file_content_view(request, pk: int):
         return FileResponse(file.file.open(), as_attachment=True, filename=file.file.name)
     except FileNotFoundError:
         raise Http404()
+
+
+@csrf_exempt
+@require_POST
+def generate_key(request: Request):
+    """Generates a key for a specific channel to use for secure communication"""
+    token = request.META.get("HTTP_AUTHORIZATION")
+    if not (token and request.body):
+        return HttpResponse("Missing token or data", status=400)
+
+    response = httpx.get(VERIFY_CONNECT_ID_URL, headers={"AUTHORIZATION": token})
+    response.raise_for_status()
+    connect_id = response.json().get("sub")
+    request_data = json.loads(request.body)
+    commcare_connect_channel_id = request_data.get("channel_id")
+    try:
+        participant_data = ParticipantData.objects.get(
+            participant__identifier=connect_id, system_metadata__commcare_connect_channel_id=commcare_connect_channel_id
+        )
+    except ParticipantData.DoesNotExist:
+        raise Http404()
+
+    if participant_data.encryption_key:
+        return JsonResponse({"key": participant_data.encryption_key})
+
+    key = base64.b64encode(secrets.token_bytes(32)).decode("utf-8")
+    participant_data.encryption_key = key
+    participant_data.save(update_fields=["encryption_key"])
+    return JsonResponse({"key": key})
```

Comment:
nit: slightly simpler option (inverted condition):
```suggestion
    if not participant_data.encryption_key:
        key = base64.b64encode(secrets.token_bytes(32)).decode("utf-8")
        participant_data.encryption_key = key
        participant_data.save(update_fields=["encryption_key"])
    
    return JsonResponse({"key": participant_data.encryption_key})
```

Reply:
Also you could put the code to create the key in the ParticipantData class: `participant_data.generate_key()`

Reply:
https://github.com/dimagi/open-chat-studio/pull/1035/commits/73c2acb064387a79ccc9dd0fe17db1ea28ba20de